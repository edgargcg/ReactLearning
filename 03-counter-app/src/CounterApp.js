import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value }) => {
    const [ counter, setCounter ] = useState(value);

    const handleSubstract = () => {
        if (counter === 0)
        return;
        
        // setCounter(counter - 1);
        setCounter(counter - 1);
    };

    const handleReset = () => setCounter(value);

    const handleAdd = () => setCounter( (c) => c + 1 );

    return (
        <React.Fragment>
            <div className="container">
                <h1>
                    CounterApp
                </h1>
                <h2>
                    { counter }
                </h2>
                <button 
                    className="container__button container__button-substract"
                    onClick={ handleSubstract }
                >
                    -
                </button>
                <button 
                    className="container__button container__button-reset"
                    onClick={ handleReset }
                >
                    Reset
                </button>
                <button 
                    className="container__button container__button-add"
                    onClick={ handleAdd }
                >
                    +
                </button>
            </div>
        </React.Fragment>        
    );
}

CounterApp.prototype = {
    value:  PropTypes.number
}

CounterApp.defaultProps = {
    value: 10
}

export default CounterApp;