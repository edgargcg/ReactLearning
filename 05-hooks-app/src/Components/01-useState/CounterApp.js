import React, { Fragment, useState } from 'react';
import './counter.css';

export const CounterApp = () => {
    const [state, setState] = useState({
        counter1: 10,
        counter2: 20
    });
    const { counter1, counter2 } = state;

    const handleAdd = () => {
        setState({
            ...state,
            counter1: counter1 + 1,
        });
    }

    return (
        <Fragment>
            <h1>
                Counters
            </h1>
            <hr />
            <div className="row">
                <h4>
                    Counter1: {counter1}
                </h4>
                <h4>
                    Counter2: {counter2}
                </h4>
            </div>
            <button 
                onClick={ handleAdd }
                className="btn btn-primary"
            >
                +
            </button>
        </Fragment>
    )
}
