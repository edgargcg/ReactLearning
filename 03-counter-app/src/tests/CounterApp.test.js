import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('Test on <CounterApp />', () => {
    let defaultValue = 100;
    const wrapper = shallow(<CounterApp />);
    const wrapperWithDefault = shallow(<CounterApp value={defaultValue} />);
    
    beforeEach(
        () => {
            defaultValue = 100;
        }
    );

    test('should render <CounterApp />', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should render <CounterApp /> with default value', () => {
        const valueShowed = wrapperWithDefault.find('h2').text();

        expect(valueShowed).toBe(`${defaultValue}`);
    })
    
    test('on <CounterApp /> should substract on click container__button-substrac button', () => {
        wrapperWithDefault.find('.container__button-substract').simulate('click');

        const result = wrapperWithDefault.find('h2').text();
        expect(result).toBe(`${defaultValue - 1}`);
    })

    test('on <CounterApp /> should reset value  on click container__button-reset button', () => {
        wrapperWithDefault.find('.container__button-substract').simulate('click');
        wrapperWithDefault.find('.container__button-substract').simulate('click');
        wrapperWithDefault.find('.container__button-substract').simulate('click');
        
        wrapperWithDefault.find('.container__button-reset').simulate('click');
        
        const result = wrapperWithDefault.find('h2').text();
        expect(result).toBe(`${defaultValue}`);
    })
    
    test('on <CounterApp /> should substract on click container__button-add button', () => {
        wrapperWithDefault.find('.container__button-add').simulate('click');

        const result = wrapperWithDefault.find('h2').text();
        expect(result).toBe(`${defaultValue + 1}`);
    })

})
