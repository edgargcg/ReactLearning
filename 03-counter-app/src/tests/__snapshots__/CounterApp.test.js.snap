// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <CounterApp /> should render <CounterApp /> 1`] = `
<Fragment>
  <div
    className="container"
  >
    <h1>
      CounterApp
    </h1>
    <h2>
      10
    </h2>
    <button
      className="container__button container__button-substract"
      onClick={[Function]}
    >
      -
    </button>
    <button
      className="container__button container__button-reset"
      onClick={[Function]}
    >
      Reset
    </button>
    <button
      className="container__button container__button-add"
      onClick={[Function]}
    >
      +
    </button>
  </div>
</Fragment>
`;
